vowpalwabbit/core/tests/igl_test.cc

```diff
@@ -0,0 +1,202 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+using namespace ::testing;
+
+#include "vw/test_common/test_common.h"
+#include "vw/core/array_parameters_dense.h"
+#include "vw/core/vw_math.h"
+#include "vw/core/merge.h"
+
+#include "vw/core/vw.h"
+#include "vw/core/parser.h"
+
+#include <gtest/gtest.h>
+
+#include <map>
+#include <iostream>
+
+using example_vector = std::vector<std::vector<std::string>>;
+using ftrl_weights_vector = std::vector<std::tuple<float, float, float, float, float, float>>;
+using separate_weights_vector = std::vector<std::tuple<size_t, float, float, float, float, float, float>>;
+
+separate_weights_vector get_separate_weights(std::unique_ptr<VW::workspace> vw) {
+  auto& weights = vw->weights.dense_weights;
+  auto iter = weights.begin();
+  auto end = weights.end();
+
+  separate_weights_vector weights_vector;
+
+  while(iter < end) {
+    bool non_zero = false;
+    for (int i=0; i < 6; i++) {
+      if (*iter[i] != 0.f) {
+        non_zero = true;
+      }
+    }
+
+    if (non_zero) {
+      weights_vector.emplace_back(iter.index_without_stride(), *iter[0], *iter[1], *iter[2], *iter[3], *iter[4], *iter[5]);
+    }
+    ++iter;
+  }
+
+  return weights_vector;
+}
+
+std::vector<separate_weights_vector> split_weights(std::unique_ptr<VW::workspace> vw) {
+  auto& weights = vw->weights.dense_weights;
+  auto iter = weights.begin();
+
+  auto end = weights.end();
+
+  separate_weights_vector sl_weights_vector;
+  separate_weights_vector multi_weights_vector;
+  std::vector<separate_weights_vector> result;
+
+  while (iter < end) {
+    bool non_zero = false;
+    for (int i=0; i < 6; i++) {
+      if (*iter[i] != 0.f) {
+        non_zero = true;
+      }
+    }
+    if (non_zero) {
+      if ((iter.index_without_stride() & (2 - 1)) == 0) {
+        sl_weights_vector.emplace_back(iter.index_without_stride()>>1, *iter[0], *iter[1], *iter[2], *iter[3], *iter[4], *iter[5]);
+      } else {
+        multi_weights_vector.emplace_back(iter.index_without_stride()>>1, *iter[0], *iter[1], *iter[2], *iter[3], *iter[4], *iter[5]);
+      }
+    }
+
+    ++iter;
+  }
+
+  result.emplace_back(sl_weights_vector);
+  result.emplace_back(multi_weights_vector);
+
+  return result;
+}
+
+TEST(IGL, ModelWeightsEqualToSeparateModelWeights) {
```

Comment:
Our usual format for test suites is PascalCase, consider `Igl`, `InteractionGround`